images/fcos-bastion-image/root/usr/lib/systemd/system/squid-proxy.service

```diff
@@ -0,0 +1,32 @@
+[Unit]
+Description=Creates the squid proxy container
+Wants=network-online.target
+After=network-online.target
+RequiresMountsFor=%t/containers
+
+[Service]
+Environment=PODMAN_SYSTEMD_UNIT=%n
+EnvironmentFile=/usr/share/bastion_services/squid/squid_systemd.conf
+Restart=on-failure
+TimeoutStopSec=70
+ExecStartPre=/bin/rm -f %t/%n.ctr-id
+ExecStartPre=/usr/bin/podman login quay.io
+ExecStartPre=-/bin/sh -c '/usr/bin/mkdir -p ${PODMAN_SQUID_BASEDIR}/{cache,log,lib}'
+ExecStartPre=-/bin/sh -c '/usr/bin/podman run --rm --interactive -v "${PODMAN_SQUID_BASEDIR}/lib:/var/lib:Z" ${PODMAN_SQUID_IMAGE} bash < /usr/share/bastion_services/squid/create_ssl_db.sh'
```

Comment:
can we run these pre-steps only when needed?


For example, we can have them running in another unit (squid-proxy-init.service) like:

```
[Unit]
Description=Prepare the squid proxy container configuration
Wants=network-online.target
After=network-online.target
RequiresMountsFor=%t/containers
ConditionPathExists=!/var/.squid-configured
[Service]
Environment=PODMAN_SYSTEMD_UNIT=%n
EnvironmentFile=/usr/share/bastion_services/squid/squid_systemd.conf
Restart=on-failure
TimeoutStopSec=70
ExecStartPre=/bin/rm -f %t/%n.ctr-id
ExecStartPre=/bin/sh -c '/usr/bin/mkdir -p ${PODMAN_SQUID_BASEDIR}/{cache,log,lib}'
ExecStartPre=/bin/sh -c '/usr/bin/podman run --rm --entrypoint=/bin/bash -v "${PODMAN_SQUID_BASEDIR}/lib:/var/lib:Z" ${PODMAN_SQUID_IMAGE} -c /usr/lib64/squid/security_file_certgen -c -s /var/lib/ssl_db -M 4'
ExecStartPre=/bin/sh -c '/usr/bin/chown -R ${PODMAN_SQUID_UID}:${PODMAN_SQUID_UID} ${PODMAN_SQUID_BASEDIR}'
ExecStartPre=/bin/sh -c 'podman unshare chown ${PODMAN_SQUID_UID}:${PODMAN_SQUID_UID} -R ${PODMAN_SQUID_BASEDIR}'
ExecStart=/bin/touch /var/.squid-configured
Type=one-shot
RemainAfterExit=no

......

```

I removed:
1. the login to quay.io as that is not needed, or did you have issues regarding it?
2. the `-` after `=` to avoid loosing failures events that we don't want to loose

Out of curiosity, why do you run `chown` twice (once via podman unshare)?


Finally, the squid-proxy.service should also have:
```
Requires=squid-proxy-init.service
After=squid-proxy-init.service
```

Consider the above an example to double check and test. Further info are available at: https://docs.fedoraproject.org/en-US/packaging-guidelines/Initial_Service_Setup/